### PR TITLE
kinematics: Add hybrid-corexy and hybrid-corexz

### PR DESCRIPTION
**Pull request by Tircown**
Add hybrid-corexy kinematics.
Add hybrid-corexz kinematics.
Both are simple carriage version of the kinematics also known as Markforged kinematics.

This follows the issue #3391 and is a revamp of pull request #3609

Signed-off-by: Fabrice GALLET tircown@gmail.com

### CODE REVIEW

File: klippy/kinematics/hybrid_corexy.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Code for handling the kinematics of hybrid-corexy robots
+# The hybrid-corexy kinematic is also known as Markforged kinematics
```
Comment (KevinOConnor) [inline klippy/kinematics/hybrid_corexy.py:2]: Please use a single line for the file description.  It's preferable for all the code to have the same format file header: 1 line comment, 1 blank line, 1 line copyright, 1 blank line, 1 line license.  You can, of course, add any amount of comment lines after the initial header block.

File: config/example-hybrid-corexy.cfg [hunk 2]
```diff
@@ -0,0 +1,79 @@
+# This file is an example config file for hybrid corexy style printers also
```
Comment (KevinOConnor) [inline config/example-hybrid-corexy.cfg:1]: When adding an "example-" config file, it's a good idea to also add it to test/klippy/printers.test regression test tool.  See https://www.klipper3d.org/Example_Configs.html

**Review by KevinOConnor — COMMENTED** [review]
Thanks.  In general it looks good to me.  I have a few comments below.

-Kevin

File: docs/Config_Reference.md [hunk 3]
```diff
@@ -365,6 +366,74 @@ max_z_accel:
...
+
+# The stepper_z section is used to describe the Z axis as well as the
+# stepper controlling the X-Z movement.
+[stepper_z]
+```
+
+## Hybrid-CoreXZ Kinematics
+
+See [example-hybrid-corexz.cfg](../config/example-hybrid-corexz.cfg)
+for an example hybrid corexz kinematics config file.
```
[the remark this answers is not in the document]
Comment (Tircown) [reply docs/Config_Reference.md:406]: Thank you for yours comments.
I have a physical mockup which allows me to test all situations for the gantry.
Ankurv is running the (full) hybrid-corexz branch on his IDEX Switchwire and print with it.
As requested on my previous PR, I split the code into more manageable sections. This PR is the for a simple carriage printer but the complement will follow soon. Thanks to Ankurv to test all of these features for real.

**Comment by KevinOConnor** [comment]
Thanks.

-Kevin

**Pull request merged**